src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+
+    # API ----------------------------------------------------------------------------
+
+    @lazyproperty
+    def shape(self):
+        return self.counts.shape
+
+    @lazyproperty
+    def zscore(self):
+        return self._calculator.zscore
+
+    @lazyproperty
+    def row_proportions(self):
+        return self._calculator.row_proportions
+
+    @lazyproperty
+    def column_proportions(self):
+        return self._calculator.column_proportions
+
+    @lazyproperty
+    def table_proportions(self):
+        return self._calculator.table_proportions
+
+    @lazyproperty
+    def counts(self):
+        return self._calculator.counts
+
+    @lazyproperty
+    def base_counts(self):
+        return self._calculator.base_counts
+
+    @lazyproperty
+    def row_margin(self):
+        return self._calculator.row_margin
+
+    @lazyproperty
+    def column_margin(self):
+        return self._calculator.column_margin
+
+    @lazyproperty
+    def table_margin(self):
+        return self._calculator.table_margin
+
+    @lazyproperty
+    def row_base(self):
+        return self._calculator.row_base
+
+    @lazyproperty
+    def column_base(self):
+        return self._calculator.column_base
+
+    @lazyproperty
+    def table_base(self):
+        return self._calculator.table_base
+
+    @lazyproperty
+    def table_percentages(self):
+        return self.table_proportions * 100
+
+    @lazyproperty
+    def column_percentages(self):
+        return self.column_proportions * 100
+
+    @lazyproperty
+    def row_percentages(self):
+        return self.row_proportions * 100
+
+    @lazyproperty
+    def population_counts(self):
+        return (
+            self.table_proportions * self._population * self._cube.population_fraction
+        )
+
+    @lazyproperty
+    def row_labels(self):
+        return self._calculator.row_labels
+
+    @lazyproperty
+    def column_labels(self):
+        return self._calculator.column_labels
+
+    # Properties ---------------------------------------------------------------------
+
+    @lazyproperty
+    def _ordering(self):
+        return (
+            OrderTransform(self._cube.dimensions, self._reordered_ids)
+            if self._reordered_ids
+            else None
+        )
+
+    @lazyproperty
+    def _transforms(self):
+        return Transforms(self._ordering, self._pruning, self._insertions)
+
+    @lazyproperty
+    def _assembler(self):
+        return Assembler(self._slice, self._transforms)
+
+    @lazyproperty
+    def _calculator(self):
+        return Calculator(self._assembler)
+
+    @lazyproperty
+    def _dimensions(self):
+        dimensions = self._cube.dimensions[-2:]
+        if self._ca_as_0th:
+            return tuple([dimensions[-1]])
+        return dimensions
+
+    @lazyproperty
+    def _insertions(self):
+        return (
+            Insertions(self._dimensions, self._slice) if self._use_insertions else None
+        )
+
+    @lazyproperty
+    def _slice(self):
```

Comment:
@scanny This will essentially be the factory method, it can probably go to what will become the `_BaseSlice`, and generate the appropriate slice from there. It still uses the cube internals, but those can be extracted (namely the part that deals with missingness and measures), after we throw away everything else (margins, percentages, and most importantly pruning).